algosdk/logic.py

```diff
@@ -0,0 +1,122 @@
+import json
+import os
+
+from . import error
+
+spec = None
+opcodes = None
+
+MAX_COST = 20000
+MAX_LENGTH = 1000
+
+def check_program(program, args=None):
+    """
+    Performs program checking for max length and cost
+
+    Args:
+        program (bytes)
+        args (list[bytes])
+
+    Returns:
+        True on success
+        Raises InvalidProgram on error
+    """
+
+    global spec, opcodes
+
+    if not program:
+        raise error.InvalidProgram("empty program")
+    if not args:
+        args = []
+
+    if spec is None:
+        script_path = os.path.realpath(__file__)
+        script_dir = os.path.dirname(script_path)
+        langspec_file = os.path.join(script_dir, "data", "langspec.json")
+        with open(langspec_file, "rt") as fin:
+            spec = json.load(fin)
```

Comment:
This won't work with the distribution package since that only includes things under the algosdk package. I think you can just move langspec.json into algosdk (something like algosdk/data/langspec.json) and then reference it with open("data/langspec.json") since relative paths should work in this case.

Reply:
I added langspec.json to setup.py, so it is in the package. that's the point, script_dir is abs path of logic.py whatever it run from. and it opens data/langspec.json that is expected to be near it.

Reply:
Oddly, this works with the wheel file but not the tar. Seems like the fix is to add a MANIFEST.in file.

Reply:
Added to MANIFEST.in as well.